### ISSUE TITLE
Add Tilt support for rapid development

### ISSUE THREAD

**Issue by johananl** [inspektor-gadget/inspektor-gadget#952]
Looking at https://github.com/kinvolk/inspektor-gadget/blob/main/docs/CONTRIBUTING.md, it seems we are currently relying on pushing images to a registry during development. This can greatly slow down development.

[Tilt](https://tilt.dev/) allows rapid development by automatically updating local code changes on a remote k8s cluster, thus saving the need to rebuild a container image, push it and re-deploy it to the cluster. It is free and open source.

P. S. Not sure if "local gadget" addresses this concern. Still, even if it does, there are probably some cases where there is no substitute for running against a real k8s cluster, in which case Tilt support can be useful.

**Comment by alban** [comment]
Good idea. Do you know how Tilt works with compiled programs? (Golang & C for the eBPF code)

Yes, local-gadget partially addresses this concern (I use it for faster local development), but as you say not everything can be tested with local-gadget.

**Comment by johananl** [comment]
>Do you know how Tilt works with compiled programs? (Golang & C for the eBPF code)

Tilt is a Python-based wrapper for whatever, so you can invoke `go` or `clang` or whatever to generate binaries.

**Comment by mauriciovasquezbernal** [comment]
As mentioned above and in https://github.com/inspektor-gadget/inspektor-gadget/pull/966#issuecomment-2011619019, we're using `ig` a lot to speed up the development, so the times we need to deploy IG to k8s are less common. I think we can live without this for now.
